### PR TITLE
alb

### PR DESCRIPTION
**Pull request by chunsanglai**
## How Has This Been Tested?
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: applicationloadbalancer/v0.0.1/main.tf [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  security_groups            = [aws_security_group.allow_lb.id]
+  subnets                    = var.subnet_ids
+  drop_invalid_header_fields = true
+
+  enable_deletion_protection = var.enable_deletion_protection
+  idle_timeout               = 3600
+  access_logs {
+    bucket  = aws_s3_bucket.lb_logs.bucket
+    prefix  = "${var.name}-lb_logs"
+    enabled = true
```
Comment (yannickvr) [inline applicationloadbalancer/v0.0.1/main.tf:15]: I'd like to see this be optional.

File: applicationloadbalancer/v0.0.1/main.tf [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+#Attach any additional certificates to the HTTPS listener
+resource "aws_lb_listener_certificate" "certificate" {
+  count           = length(var.additional_certificate_arns)
+  listener_arn    = aws_lb_listener.https.arn
+  certificate_arn = var.additional_certificate_arns[count.index]
+}
+
+#Create the public security group
+data "aws_ec2_managed_prefix_list" "cloudfront" {
```
Comment (yannickvr) [inline applicationloadbalancer/v0.0.1/main.tf:61]: This config will only allow use of this resource together with CloudFront. This should be optional, so the Ingress rules should be configurable

File: applicationloadbalancer/v0.0.1/main.tf [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  ingress {
+    description     = "443 from internet"
+    from_port       = 443
+    to_port         = 443
+    protocol        = "tcp"
+    prefix_list_ids = [data.aws_ec2_managed_prefix_list.cloudfront.id]
+  }
+}
+
+resource "aws_s3_bucket" "lb_logs" {
```
Comment (yannickvr) [inline applicationloadbalancer/v0.0.1/main.tf:79]: also these access log features should be optional

File: applicationloadbalancer/v0.0.1/main.tf (hunk 1 above)
Comment (chunsanglai) [reply applicationloadbalancer/v0.0.1/main.tf:15]: This whole access_logs block of code? @yannickvr

Comment (yannickvr) [reply applicationloadbalancer/v0.0.1/main.tf:15]: Anything that's got to do with access logs. Not all environments require access logs to be enabled, so if we disable that, all features need to be disabled as well.

**Comment by yannickvr** [comment]
needs rafactor

**Pull request closed without merging**